src/VisualStudio/Core/Def/Snippets/SnippetExpansionClient.cs

```diff
@@ -739,13 +744,17 @@ private void OnModelUpdated(object sender, ModelUpdatedEventsArgs e)
         // TODO: The following blocks the UI thread without cancellation, but it only occurs when an argument value
         // completion session is active, which is behind an experimental feature flag.
         // https://github.com/dotnet/roslyn/issues/50634
-        var compilation = _threadingContext.JoinableTaskFactory.Run(() => document.Project.GetRequiredCompilationAsync(CancellationToken.None));
-        var newSymbolKey = (e.NewModel.SelectedItem as AbstractSignatureHelpProvider.SymbolKeySignatureHelpItem)?.SymbolKey ?? default;
-        var newSymbol = newSymbolKey.Resolve(compilation, cancellationToken: CancellationToken.None).GetAnySymbol();
+        var compilation = await document.Project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(true);
+        var newSymbolKey = (newModel.SelectedItem as AbstractSignatureHelpProvider.SymbolKeySignatureHelpItem)?.SymbolKey ?? default;
+        var newSymbol = newSymbolKey.Resolve(compilation, cancellationToken: cancellationToken).GetAnySymbol();
         if (newSymbol is not IMethodSymbol method)
             return;
 
-        MoveToSpecificMethod(method, CancellationToken.None);
+        await MoveToSpecificMethodAsync(
+            document, method, cancellationToken).ConfigureAwait(true);
+
+        // Don't let the compilation drop as MoveToSpecificMethodAsync wants to get the semantic model for the document.
+        GC.KeepAlive(compilation);
```

Comment:
Does this `KeepAlive` need to be moved? Wouldn't `MoveToSpecificMethodAsync` be complete by this point?

Reply:
I don't understand the question.  The point of the keepalive is to say "keep the value in memor from teh point of computation to this point, so it is not GCed".

Reply:
Gotcha, a misunderstanding on my part. Why this works this way totally makes sense now. =)